### PR TITLE
devcontainer: use fish shell

### PR DESCRIPTION
**Pull request by DerekNonGeneric**

### CODE REVIEW

File: .devcontainer/Dockerfile [hunk 1]
```diff
@@ -8,19 +8,8 @@ ENV LANG=en_US.UTF-8 \
...
-ARG NODE_VERSION="lts/*"
-RUN su vscode -c "source /usr/local/share/nvm/nvm.sh && nvm install ${NODE_VERSION} 2>&1"
-
 # [Optional] Uncomment this section to install additional OS packages.
 RUN apt-get update && export DEBIAN_FRONTEND=noninteractive \
-    && apt-get -y install --no-install-recommends gnupg2 default-jre
-
-# [Optional] Uncomment this line to install additional gems.
-# RUN gem install <your-gem-names-here>
+    && apt-get -y install --no-install-recommends fish gnupg2 default-jre
```
Comment (DerekNonGeneric) [inline .devcontainer/Dockerfile:13]: fish installed here

File: .devcontainer/Dockerfile [hunk 2]
```diff
@@ -8,19 +8,8 @@ ENV LANG=en_US.UTF-8 \
...
 
-# [Optional] Uncomment this line to install global node packages.
-# RUN su vscode -c "source /usr/local/share/nvm/nvm.sh && npm install -g <your-package-here>" 2>&1
+RUN fish -c 'curl -sL https://raw.githubusercontent.com/jorgebucaran/fisher/main/fisher.fish | source && fisher install jorgebucaran/{fisher,nvm.fish}'
```
Comment (DerekNonGeneric) [inline .devcontainer/Dockerfile:15]: fisher and plugins installed here

File: .devcontainer/devcontainer.json [hunk 3]
```diff
@@ -64,7 +64,7 @@
   ],
 
   // Use 'postCreateCommand' to run commands after the container is created.
-  "postCreateCommand": "sh .devcontainer/post-create.sh",
+  // "postCreateCommand": "sh .devcontainer/post-create.sh",
```
Comment (DerekNonGeneric) [inline .devcontainer/devcontainer.json:67]: ignoring this for now

File: .devcontainer/Dockerfile [hunk 4]
```diff
@@ -8,19 +8,33 @@ ENV LANG=en_US.UTF-8 \
...
-# [Optional] Uncomment this line to install additional gems.
-# RUN gem install <your-gem-names-here>
+RUN mkdir /usr/share/man/man1/
+
+RUN apt-get update && export DEBIAN_FRONTEND=noninteractive \
+    && apt-get -y install --no-install-recommends fish default-jre
 
-# [Optional] Uncomment this line to install global node packages.
-# RUN su vscode -c "source /usr/local/share/nvm/nvm.sh && npm install -g <your-package-here>" 2>&1
+RUN fish -c 'curl -sL https://raw.githubusercontent.com/jorgebucaran/fisher/main/fisher.fish | source && fisher install jorgebucaran/{fisher,nvm.fish}'
```
Comment (jorgebucaran) [inline .devcontainer/Dockerfile]: No need for `L` if you're going to use the raw URL.

File: .devcontainer/library-scripts/common-debian.sh [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+    USER_GID=0
+fi
+
+# Load markers to see which steps have already run
+MARKER_FILE="/usr/local/etc/vscode-dev-containers/common"
+if [ -f "${MARKER_FILE}" ]; then
+    echo "Marker file found:"
+    cat "${MARKER_FILE}"
+    source "${MARKER_FILE}"
+fi
```
Comment (DerekNonGeneric) [inline .devcontainer/library-scripts/common-debian.sh]:
```suggestion
```

This is unnecessary.

File: .devcontainer/library-scripts/common-debian.sh [hunk 6]
```diff
@@ -0,0 +1,188 @@
...
+        groupadd --gid $USER_GID $USERNAME
+    fi
+    if [ "${USER_UID}" = "automatic" ]; then
+        useradd -s /bin/bash --gid $USERNAME -m $USERNAME
+    else
+        useradd -s /bin/bash --uid $USER_UID --gid $USERNAME -m $USERNAME
+    fi
+fi
+
+# Add add sudo support for non-root user
```
Comment (DerekNonGeneric) [inline .devcontainer/library-scripts/common-debian.sh]:
```suggestion
# Add sudo support for non-root user
```

**Pull request merged**